src/main/scala/exercise3/KarplusStrong.scala

```diff
@@ -0,0 +1,54 @@
+package exercise3
+
+import exercise2._
+import scala.util.Success
+import scala.util.Failure
+import exercise2.lib.StdAudio
+import zio.{UIO, URIO, ZIO, App}
+import zio.random._
+
+object KarplusStrong extends App {
+
+  def run(args: List[String]) =
+    (for {
+      noise <- whiteNoise()
+      _ <- loop(noise)
+    } yield ()).exitCode
+
+  def play(sample: Double): UIO[Unit] = ZIO.effectTotal(StdAudio.play(sample))
+
+  def whiteNoise(
+      frequency: Int = 440,
+      volume: Double = 1.0
+  ): URIO[Random, Queue[Double]] = {
+    if (frequency <= 0 && volume < 0.0 && volume > 1.0)
+      throw new Throwable("Incorrect parameters")
```

Comment:
Really not nice, you say your method can't fail: ```URIO[Random, Queue[Double]]``` but then throw an exception. Encode as ```ZIO[Random, Throwable, Queue[Double]]``` and use ```ZIO.fail(new Throwable("Incorrect parameters"))``` instead!